install.sh

```diff
@@ -175,6 +174,38 @@ function install() {
     fi
 }
 
+function installEctool() {
+    echo "installing ectool"
+
+    TEMP_FOLDER='./.temp'
+    rm -rf "$TEMP_FOLDER"
+    mkdir "$TEMP_FOLDER"
+
+    ectoolJobId="$(cat './fetch/ectool/linux/gitlab_job_id')"
+    ectoolSha256Hash="$(cat './fetch/ectool/linux/hash.sha256')"
+
+    artifactsZipFile="$TEMP_FOLDER/artifact.zip"
+
+    echo "downloading artifact from gitlab"
+    curl -s -o "$artifactsZipFile" -L "https://gitlab.howett.net/DHowett/ectool/-/jobs/${ectoolJobId}/artifacts/download?file_type=archive" || return 1
+
+    echo "checking artifact sha256 sum"
+    actualEctoolSha256Hash=$(sha256sum "$artifactsZipFile" | cut -d ' ' -f 1)
+    if [[ "$actualEctoolSha256Hash" != "$ectoolSha256Hash" ]]; then
+        echo "Incorrect sha256 sum for ectool gitlab artifact N�$ectoolJobId : $ectoolSha256Hash != $actualEctoolSha256Hash"
+        exit 1
+    fi
+
+    echo "extracting artifact"
+    unzip -q -j "$artifactsZipFile" '_build/src/ectool' -d "$TEMP_FOLDER" || return 1
+
+    cp "$TEMP_FOLDER/ectool" "$DEST_DIR$PREFIX_DIR/bin/ectool" || return 1
+    chmod +x "$DEST_DIR$PREFIX_DIR/bin/ectool" || return 1
+    rm -rf "$TEMP_FOLDER"
```

Comment:
It would be nice the `TEMP_FOLDER` removal was done outside of the function, so that in the case where the function fails, the folder is still cleaned.
Maybe by handing `installEctool` the path to create such a folder and then chaining its removal at line 128 ?